ambari-server/src/main/java/org/apache/ambari/server/security/encryption/AESEncryptionService.java

```diff
@@ -76,9 +78,11 @@ private final String getAmbariMasterKey() {
 
   private void initEnvironmentMasterKeyService() {
     if (environmentMasterKeyService == null) {
-      environmentMasterKeyService = new MasterKeyServiceImpl();
-      if (!environmentMasterKeyService.isMasterKeyInitialized()) {
-        throw new SecurityException("You are trying to use a persisted master key but its initialization has been failed!");
+      Configuration configuration = new Configuration();
+      try {
+        environmentMasterKeyService = MasterKeyServiceImpl.getMasterKeyService(null, configuration.getMasterKeyLocation(), configuration.isMasterKeyPersisted());
```

Comment:
What if Configuration is injected into MasterKeyServiceImpl? In theory this service needs to know where to look after the master key location and related stuff.